File: src/main/java/care/solve/protocol/schedule/service/ScheduleService.java
package care.solve.protocol.schedule.service;

import care.solve.protocol.schedule.entity.Schedule;
import care.solve.protocol.schedule.entity.Slot;
import org.hyperledger.fabric.sdk.exception.ChaincodeEndorsementPolicyParseException;

import java.io.IOException;

public interface ScheduleService {
    Schedule createSchedule(Schedule schedule);

    Slot createSlot(String scheduleId, Slot slot);

    void updateSlot(String scheduleId, String slotId, Slot slot);

    Schedule getSchedule(String ownerId) throws IOException, ChaincodeEndorsementPolicyParseException;
}

File: src/main/go/src/care.solve.schedule/protocol/schedule_service.go
package protocol

import (
	careproto "./proto"
	"github.com/hyperledger/fabric/core/chaincode/shim"
)

type ScheduleService interface {

	GetScheduleByOwnerId(stub shim.ChaincodeStubInterface, ownerId string) (*careproto.Schedule, error)
	CreateSchedule(stub shim.ChaincodeStubInterface, schedule careproto.Schedule) (*careproto.Schedule, error)
	CreateSlot(stub shim.ChaincodeStubInterface, scheduleId string, slot careproto.Slot) (*careproto.Slot, error)
	UpdateSlot(stub shim.ChaincodeStubInterface, scheduleId string, slotId string, newSlot careproto.Slot) error
	DecodeScheduleByteString(scheduleByteString string) (*careproto.Schedule, error)
	DecodeSlotByteString(scheduleByteString string) (*careproto.Slot, error)

}